gufe/storage/externalresource/filestorage.py

```diff
@@ -20,6 +20,12 @@ def __init__(self, root_dir: Union[pathlib.Path, str]):
     def _exists(self, location):
         return self._as_path(location).exists()
 
+    def __eq__(self, other):
+        return (
+            isinstance(other, FileStorage)
```

Comment:
a bit nickpicky, but technically objects are meant to return `NotImplemented` in this case, so that the other object can have a shot at defining `__eq__`, see https://stackoverflow.com/a/42588737